tests/manage/z_cluster/cluster_expansion/test_delete_pod.py

```diff
@@ -0,0 +1,135 @@
+            for d in TimeoutSampler(
+                timeout=timeout, sleep=2, func=d.set_resource, resource=resource_name
+            ):
+                d.delete_resource(resource_id)
+                if self.new_pods_in_status_running:
+                    logging.info("New osd pods in status running")
+                    break
+        except TimeoutExpiredError:
+            logging.warning(
+                f"New osd pods are not in status running after {timeout} seconds"
+            )
+
+    def add_capacity_with_resource_delete(self, resource_name, resource_id, is_kill_resource_repeatedly=False):
+        used_percentage = get_percent_used_capacity()
+        logging.info(f"storageutilization is completed. used capacity = {used_percentage}")
+
+        max_osds = 15
+        osd_pods_before = pod_helpers.get_osd_pods()
+        number_of_osd_pods_before = len(osd_pods_before)
+        if number_of_osd_pods_before >= max_osds:
+            pytest.skip("We have maximum of osd's in the cluster")
+
+        d = Disruptions()
+        d.set_resource(resource_name)
+
+        self.new_pods_in_status_running = False
+        osd_size = storage_cluster.get_osd_size()
+        logging.info(f"Adding one new set of OSDs. osd size = {osd_size}")
+        storagedeviceset_count = storage_cluster.add_capacity(osd_size)
+        logging.info("Adding one new set of OSDs was issued without problems")
+
+        pod_helpers.wait_for_new_osd_pods_to_come_up(number_of_osd_pods_before)
+        logging.info(f"Delete a {resource_name} pod while storage capacity is getting increased")
+        if is_kill_resource_repeatedly:
+            logging.info("kill resource repeatedly chosen")
+            with ThreadPoolExecutor() as executor:
+                executor.submit(self.kill_resource_repeatedly, resource_name, resource_id)
+        else:
+            d.delete_resource(resource_id)
+
+        pod = OCP(
+            kind=constants.POD, namespace=config.ENV_DATA['cluster_namespace']
+        )
+
+        pod.wait_for_resource(
+            timeout=420,
+            condition=constants.STATUS_RUNNING,
+            selector='app=rook-ceph-osd',
+            resource_count=storagedeviceset_count * 3
+        )
+
+        self.new_pods_in_status_running = True
+        logging.info("Finished verifying add capacity when one of the osd pods gets deleted")
+        logging.info("Waiting for ceph health check to finished...")
+        ceph_health_check(
+            namespace=config.ENV_DATA['cluster_namespace'], tries=80
+        )
+
+
+@pytest.mark.parametrize(
+    argnames=["workload_storageutilization_rbd", "resource_name", "resource_id"],
+    argvalues=[
+        pytest.param(
+            *[(0.11, True, 120), constants.ROOK_OPERATOR, 0],
+            marks=pytest.mark.polarion_id("OCS-1206")
+        ),
+    ],
+    indirect=["workload_storageutilization_rbd"],
+)
+@ignore_leftovers
+@tier4a
+class TestAddCapacityRookOperatorPodDelete(ManageTest):
```

Comment:
instead of multiple classes, one for each resource type, consider using pytest parametrize.
See reference in https://github.com/red-hat-storage/ocs-ci/blob/cee261845a6f59019d429b2f8b8f94d7315a665e/tests/manage/pv_services/test_pvc_disruptive.py#L25

Reply:
You can cosolidate https://github.com/red-hat-storage/ocs-ci/blob/master/tests/manage/z_cluster/cluster_expansion/test_add_capacity_with_node_restart.py into this test

Reply:
Yes, that makes sense. But the thing is we need to divide it also to multiple Polarion test cases. And I am not sure if it is possible with pytest parametrize

Reply:
Actually now I see this test https://github.com/red-hat-storage/ocs-ci/blob/master/tests/manage/mcg/test_bucket_creation.py that uses multiple Polarion cases. So I will use it as a reference.